src/main/java/com/example/petbuddybackend/service/user/CaretakerService.java

```diff
@@ -97,27 +107,82 @@ public RatingResponse deleteRating(String caretakerEmail, String clientEmail) {
+        List<PhotoLink> uploadedOfferPhotos = photoService.uploadPhotos(newOfferPhotos);
+        Caretaker caretaker = caretakerMapper.mapToCaretaker(createCaretakerDTO, appUser, uploadedOfferPhotos);
 
-        renewCaretakerProfilePicture(caretaker);
+        renewCaretakerPictures(caretaker);
         return caretakerMapper.mapToCaretakerComplexInfoDTO(caretakerRepository.save(caretaker));
     }
 
-    public CaretakerComplexInfoDTO editCaretaker(ModifyCaretakerDTO caretakerDTO, String email) {
-        AppUser appUser = userService.getAppUser(email);
-        Caretaker caretaker = getCaretakerByEmail(appUser.getEmail());
+    @Transactional
+    public CaretakerComplexInfoDTO editCaretaker(
+            ModifyCaretakerDTO modifyCaretakerDTO,
+            String email,
+            List<MultipartFile> newOfferPhotos
+    ) {
+        Caretaker caretaker = getCaretakerByEmail(email);
+        caretakerMapper.updateCaretakerFromDTO(caretaker, modifyCaretakerDTO);
 
-        renewCaretakerProfilePicture(caretaker);
-        caretakerMapper.updateCaretakerFromDTO(caretakerDTO, caretaker);
+        applyOfferPhotosPatch(caretaker, modifyCaretakerDTO.offerBlobsToKeep(), newOfferPhotos);
+        renewCaretakerPictures(caretaker);
         return caretakerMapper.mapToCaretakerComplexInfoDTO(caretakerRepository.save(caretaker));
     }
 
-    private Caretaker renewCaretakerProfilePicture(Caretaker caretaker) {
-        userService.renewProfilePicture(caretaker.getAccountData());
-        return caretaker;
+    @Transactional
+    public List<PhotoLinkDTO> patchOfferPhotos(
+            String email,
+            Set<String> offerBlobsToKeep,
+            List<MultipartFile> newOfferPhotos
+    ) {
+        Caretaker caretaker = getCaretakerByEmail(email);
+        applyOfferPhotosPatch(caretaker, offerBlobsToKeep, newOfferPhotos);
+        renewCaretakerPictures(caretaker);
+
+        return caretakerRepository.save(caretaker).getOfferPhotos().stream()
+                .map(photoMapper::mapToPhotoLinkDTO)
+                .toList();
+    }
+
+    private void applyOfferPhotosPatch(Caretaker caretaker, Set<String> blobsToKeep, List<MultipartFile> newPhotos) {
+        List<PhotoLink> currentPhotos = caretaker.getOfferPhotos();
+
+        int currentPhotosSize = currentPhotos.size();
+        int photosToKeepSize = Math.min(currentPhotosSize, blobsToKeep.size() + newPhotos.size());
+        int photosToRemoveSize = Math.max(0, currentPhotosSize - photosToKeepSize);
+
+        List<PhotoLink> photosToKeep = new ArrayList<>(photosToKeepSize);
+        List<PhotoLink> photosToRemove = new ArrayList<>(photosToRemoveSize);
+
+        currentPhotos.forEach(photo -> {
+            if(blobsToKeep.contains(photo.getBlob())) {
+                photosToKeep.add(photo);
+            } else {
+                photosToRemove.add(photo);
+            }
+        });
+
+        caretaker.getOfferPhotos().clear();
```

Comment:
Jest to eleganckie rozwiazanie, ale duzo dzialan na listach jest robione przez to, ze czyscimy na poczatku.
Moze najpierw usunac z GetOfferPhotos zdjecia w photosToRemove, a potem poscic addAll na newPhotos?

Reply:
```suggestion
            private void applyOfferPhotosPatch(Caretaker caretaker, Set<String> blobsToKeep, List<MultipartFile> newPhotos) {
        List<PhotoLink> currentPhotos = caretaker.getOfferPhotos();

        int currentPhotosSize = currentPhotos.size();
        int photosToKeepSize = Math.min(currentPhotosSize, blobsToKeep.size() + newPhotos.size());
        int photosToRemoveSize = Math.max(0, currentPhotosSize - photosToKeepSize);

        List<PhotoLink> photosToRemove = new ArrayList<>(photosToRemoveSize);

        currentPhotos.forEach(photo -> {
            if(!blobsToKeep.contains(photo.getBlob())) {
                photosToRemove.add(photo);
            }
        });

        caretaker.getOfferPhotos().removeAll(photosToRemove);
        caretaker.getOfferPhotos().addAll(photoService.uploadPhotos(newPhotos));
        photoService.deletePhotos(photosToRemove);

    }
```